cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericRowAggregator.java

```diff
@@ -88,21 +88,21 @@ public void aggregate(NGSIEvent event) {
             LOGGER.debug("[" + getName() + "] Processing context attribute (name=" + attrName + ", type="
                     + attrType + ")");
             // aggregate the attribute information
-            aggregation.get(NGSIConstants.RECV_TIME_TS).add(new JsonPrimitive(Long.toString(recvTimeTs)));
+            aggregation.get(NGSIConstants.RECV_TIME_TS).add(new JsonPrimitive(Long.toString(getRecvTimeTsValue(event))));
             aggregation.get(NGSIConstants.RECV_TIME).add(new JsonPrimitive(recvTime));
             aggregation.get(NGSIConstants.FIWARE_SERVICE_PATH).add(new JsonPrimitive(getServicePathForData()));
             aggregation.get(NGSIConstants.ENTITY_ID).add(new JsonPrimitive(entityId));
             aggregation.get(NGSIConstants.ENTITY_TYPE).add(new JsonPrimitive(entityType));
             aggregation.get(NGSIConstants.ATTR_NAME).add(new JsonPrimitive(attrName));
             aggregation.get(NGSIConstants.ATTR_TYPE).add(new JsonPrimitive(attrType));
-            aggregation.get(NGSIConstants.ATTR_VALUE).add(attrValue);
+            aggregation.get(NGSIConstants.ATTR_VALUE).add(adaptAttrValue(attrValue));
```

Comment:
This shouldn't be here. This data is very specific for this solution. This is a generic class which is used by many sinks.